client/src/lifecycleManager.ts

```diff
@@ -56,69 +78,79 @@ class LifecycleManager extends LuigiClientBase {
    * @example
    * LuigiClient.luigiClientInit()
    */
-  luigiClientInit() {
+  luigiClientInit(): void {
     if (this.luigiInitialized) {
       console.warn('Luigi Client has been already initialized');
       return;
     }
+
     /**
      * Save context data every time navigation to a different node happens
      * @private
      */
-    const setContext = rawData => {
+    const setContext = (rawData: any) => {
       for (let index = 0; index < this.defaultContextKeys.length; index++) {
-        let key = this.defaultContextKeys[index];
+        let key: string = this.defaultContextKeys[index];
+
         try {
           if (typeof rawData[key] === 'string') {
             rawData[key] = JSON.parse(rawData[key]);
           }
-        } catch (e) {
-          console.info('unable to parse luigi context data for', key, rawData[key], e);
+        } catch (error) {
+          console.info('unable to parse luigi context data for', key, rawData[key], error);
         }
       }
       this.setCurrentContext(rawData);
     };
 
-    const setAuthData = eventPayload => {
+    const setAuthData = (eventPayload: any) => {
       if (eventPayload) {
         this.authData = eventPayload;
       }
     };
 
-    helpers.addEventListener('luigi.init', e => {
-      setContext(e.data);
-      setAuthData(e.data.authData);
-      helpers.setLuigiCoreDomain(e.origin);
+    helpers.addEventListener('luigi.init', (event: any) => {
+      setContext(event.data);
+      setAuthData(event.data.authData);
+      helpers.setLuigiCoreDomain(event.origin);
       this.luigiInitialized = true;
-      this._notifyInit(e.origin);
+      this._notifyInit(event.origin);
       helpers.sendPostMessageToLuigiCore({ msg: 'luigi.init.ok' });
     });
 
-    helpers.addEventListener('luigi-client.inactive-microfrontend', e => {
-      this._notifyInactive(e.origin);
+    helpers.addEventListener('luigi-client.inactive-microfrontend', (event: any) => {
+      this._notifyInactive(event.origin);
     });
 
-    helpers.addEventListener('luigi.auth.tokenIssued', e => {
-      setAuthData(e.data.authData);
+    helpers.addEventListener('luigi.auth.tokenIssued', (event: any) => {
+      setAuthData(event.data.authData);
     });
 
-    helpers.addEventListener('luigi.navigate', e => {
-      setContext(e.data);
-      if (!this.currentContext.internal.isNavigateBack && !this.currentContext.withoutSync) {
-        const previousHash = window.location.hash;
-        history.replaceState({ luigiInduced: true }, '', e.data.viewUrl);
+    helpers.addEventListener('luigi.navigate', (event: any) => {
+      setContext(event.data);
+
+      if (
+        !(this.currentContext.internal as Record<string, any>)['isNavigateBack'] &&
+        !this.currentContext['withoutSync']
```

Comment:
not sure about this line. will need to revisit it

Reply:
This is because both 'isNavigateBack' and 'withoutSync' are not explicitly defined in optional 'internal' prop